classes/blocks/class-articles.php

```diff
@@ -314,17 +314,25 @@ function ( $tag ) {
 		// Get all posts with arguments.
 		$args = self::DEFAULT_POST_ARGS;
 
+		// For post page block, so current main post will exclude.
+		if ( $exclude_post_id ) {
+			$args['post__not_in'] = [ $exclude_post_id ];
+
+			// Get current post tags & categories, if not pass.
+			if ( ! $tags ) {
```

Comment:
Hi @Inwerpsel , I like your idea of adding a new field(checkbox of "inherit filters from post") to offer a more control on articles block on post pagetype(cause this issue only happens on post & not on other post types), but without adding this field the current implementation work as it was working before this change.

I just thought I have to highlight that, if editor wants to add a custom taxonomy filters, then they can easily able to add it, and if we introduce a new field here it will bit confuse editor about behaviour of articles block with and without this field, also I don't think the unchecked checkbox of "inherit filters from post" will be more useful cause mostly the goal is to display the related articles in articles block, so if the taxonomy filter(tags & categories) is empty in articles block then the articles block shows the posts by filtering on current post taxonomy will be the default behaviour.

I may miss some aspects of your suggestion, so I thought I put my thoughts here.

cc: @comzeradd , @suzi-greenpeace

Reply:
@sagarsdeshmukh To be sure I understand correctly, you suggest to make inheriting the filters implicit each time an Articles block has no filters?

That makes some sense, as there's likely no use case for Articles with no filters. Would still be nice to include a notice in the editor about this behavior.

For example if a post has 2 categories and 3 tags. If an editor would then add 1 tag manually to the block, it would stop using the other tags and categories and only filter the manual query. That's fine but if we don't describe the behavior it could be confusing.

I assume we also need to account for that in the block editor code? I mean, with the current behavior the post's current tags and categories are set on the block by our back end, so only when you view the post. If we want consistent behavior in the editor, it would need to update the block every time the post's tags or categories are changed.

Reply:
Hi @Inwerpsel 
> you suggest to make inheriting the filters implicit each time an Articles block has no filters?

yes,

> That makes some sense, as there's likely no use case for Articles with no filters. Would still be nice to include a notice in the editor about this behavior.

The note added as below,
![image](https://user-images.githubusercontent.com/5357471/159247485-a7401f21-be6d-4bc7-8ba9-4358431c3672.png)


> For example if a post has 2 categories and 3 tags. If an editor would then add 1 tag manually to the block, it would 
stop using the other tags **and** categories and only filter the manual query.

Here, it only stop using post tags and use custom tag in filter, but categories filter will still remain as post categories if available.

> I assume we also need to account for that in the block editor code? I mean, with the current behavior the post's current tags and categories are set on the block by our back end, so only when you view the post. If we want consistent behavior in the editor, it would need to update the block every time the post's tags or categories are changed.

That's true, but categories & tags are not frequently changes, but if changes the articles block not refresh automatically till editor refresh the page.

Reply:
> but if changes the articles block not refresh automatically till editor refresh the page

Seems acceptable for a first iteration. But should be possible to fix that using WP's JS data API, we could do a follow up for that.

For example using
```js
const tags = useSelect( select => {
  return select('core/editor').getCurrentPost().tags;
}
```

We could force the block to rerender